src/components/wizard.component.jsx

```diff
@@ -43,6 +43,11 @@ export default class Wizard extends React.PureComponent {
       event.preventDefault();
     }
     if (index < 0 || index > this.props.steps.length - 1) return;
+
+    if (this.props.onStepChange && (this.state.currentStep !== index)) {
+      this.props.onStepChange(this.state.currentStep, index);
```

Comment:
It would be good to have in onStepChange callback function step.id as one of the parameters. It could be useful in UI to operate also with steps ids, not only with indexes